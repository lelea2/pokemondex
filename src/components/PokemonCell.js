import React, { Component } from 'react';
import './styles/PokemonCell.css';
import PropTypes from 'prop-types';
import clss from 'classnames';

let img = `https://img.pokemondb.net/artwork/@@@NAME@@@.jpg`;

class PokemonCell extends Component {

  getDetail() {
    const { url } = this.props;
    this.props.getDetail(url);
  }

  render() {
    const { name, active } = this.props;
    const imgURL = img.replace('@@@NAME@@@', name);
    return (
      <button
        className={clss("poke-cell", { active })}
        onClick={this.getDetail.bind(this)}
      >
        <img src={imgURL} alt={name} width="50" height="50" />
      </button>
    );
  }
}

PokemonCell.propTypes = {
  name: PropTypes.string,
  url: PropTypes.string,
  active: PropTypes.bool,
  getDetail: PropTypes.func
};

export default PokemonCell;
