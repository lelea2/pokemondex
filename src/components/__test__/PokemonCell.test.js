import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import PokemonCell from '../PokemonCell';

describe('Component: PokemonCell', () => {

  it("PokemonCell renders correctly", () => {
    const test = renderer.create(<PokemonCell name="Pikachu" url="" active={false} getDetail={() => null} />).toJSON();
    expect(test).toMatchSnapshot();
  });

  it("PokemonCell - active renders correctly", () => {
    const test = renderer.create(<PokemonCell name="Pikachu" url="" active={true} getDetail={() => null} />).toJSON();
    expect(test).toMatchSnapshot();
  });

  it("PokemonCell - onClick", () => {
    const event = {
      preventDefault() {}
    };
    const getDetailMock = jest.fn();
    const component = shallow(<PokemonCell name="Pikachu" url="http://test.com/pikachu" active={true} getDetail={getDetailMock} />);
    component.find('button').simulate('click', event);
    expect(getDetailMock).toHaveBeenCalledWith('http://test.com/pikachu');
  });

});