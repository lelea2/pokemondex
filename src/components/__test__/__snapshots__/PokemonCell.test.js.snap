// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: PokemonCell PokemonCell - active renders correctly 1`] = `
<button
  className="poke-cell active"
  onClick={[Function]}
>
  <img
    alt="Pikachu"
    height="50"
    src="https://img.pokemondb.net/artwork/Pikachu.jpg"
    width="50"
  />
</button>
`;

exports[`Component: PokemonCell PokemonCell renders correctly 1`] = `
<button
  className="poke-cell"
  onClick={[Function]}
>
  <img
    alt="Pikachu"
    height="50"
    src="https://img.pokemondb.net/artwork/Pikachu.jpg"
    width="50"
  />
</button>
`;
